### DISCUSSION TITLE
Projekt Überschussladen

### DISCUSSION THREAD

**Discussion by Call777** [camueller/SmartApplianceEnabler#266] [Installation und Betrieb]
Hallo,
ich heiße Stefan und ich möchte mehr Strom aus meiner PV Anlage nutzen, deshalb würde ich gerne folgendes Projekt umsetzten.
Ein Schichtpeicher mit 2 stufenlosen, regelbarer Heizstäben und einer Leistung von jeweils maximal 3KW soll bei Überproduktion der PV Anlage das Wasser erhitzen.
Ebenso soll ein E-Auto bei Überproduktion geladen werden.
Das Ganze am besten Vollautomatisch oder bei Bedarf manuell steuerbar. Ich habe bereits Kontakt zu einer Installationsfirma aufgenommen und basieren auf meinem Wechselrichter von SMA wurde auch eine Lösung, die auf dem Home Manager aufbaut, aufgezeigt.
Die Firma baut auf dem Übertragungsprotokoll  SEMP auf, was die Auswahl der Hardware (Heizstab und Wallbox) doch sehr einschränkt und zudem noch recht teuer ist.
Bezüglich des E-Autos wird sich die Hardware in absehbarer Zeit wieder ändern, weil ich auf das Bidirektionale Laden von VW setzte, was auch ausschlaggebend  für den Kauf des Fahrzeugs war.
VW hat mir zugesichert das dies 2022 startet aber ein ganz anderes Equipment erfordert als es heute am Markt üblich ist.
Da ich nicht weiß wann es schlussendlich losgeht und diese Hardware kosten wird, das E-Auto aber im August/September eintrifft,  suche ich nach einer Übergangslösung fürs Auto und einer Lösung für die Heizstäbe.
Da bin ich auf dieses Forum gestoßen und es ließt sich so, als sollte das funktionieren.
Den Home Manager von SMA werde ich auf jeden Fall einbauen lassen.
Ich bin leider, was Linux angeht, vollkommener Laie und bin mir auch nicht sicher ob ich das so ans Laufen bekomme wie einige hier im Forum.
Benötigen tue ich als erstes wahrscheinlich ein Raspberry, mein Nas von Synology kann die Aufgabe nicht übernehmen?
Die Ersteinrichtung erfolgt wahrscheinlich direkt über das Rapberry, kann man nach der Installation dann über das Netzwerk darauf zugreifen oder muss man immer vor Ort an dem Raspberry arbeiten?
Als Heizstab ginge ja dann ein Model was PWM gesteuert ist und direkt an das Raspberry angeschlossen wird?
Welche Wallbox wäre am geeignetsten um stufenlos zu laden? 
Wie gesagt bin ich kein PC Neuling aber habe doch Respekt vor diesem Projekt und möchte Fragen ob ich hier im Forum Unterstützung bekommen kann oder es besser lassen sollte?
Wenn jemand aus dem Postleitzahlengebiet 666 hier unterwegs wäre und sich damit auskennt wäre das natürlich super.
Ich hätte kein Problem eine Firma damit zu beauftragen aber erstens ist es sehr schwierig überhaupt jemanden zu finden, der dieses Jahr noch Zeit hat aber was viel schwieriger ist, das sich jemand mit solchen Steuerungen auskennt.
Vielen Dank vorab und Gruß Stefan.

**Comment by camueller** [comment]
Hallo Stefan,

> Ein Schichtpeicher mit 2 stufenlosen, regelbarer Heizstäben und einer Leistung von jeweils maximal 3KW soll bei Überproduktion der PV Anlage das Wasser erhitzen.

Mittels PWM oder mehreren Schaltern sollte das gehen.

>Ebenso soll ein E-Auto bei Überproduktion geladen werden.

Sollte kein Problem sein, wenn Du eine unterstütze Wallbox verwendest.

>Ich bin leider, was Linux angeht, vollkommener Laie und bin mir auch nicht sicher ob ich das so ans Laufen bekomme wie einige hier im Forum.

Irgenwann waren wir alle mal Laien ... 
Wie bei (fast) jedem Problem gibt es zwei Lösungen: Entweder macht man sich selbst schlau (auch mit Hilfe von Communities im Internet) oder man gibt jemandem Geld dafür, dass er das Problem löst. Nachteil des Letzteren ist, dass so ein Thema nicht zu einem Zeitpunkt abgeschlossen ist, sondern möglicherweise immer wieder Hand angelegt werden muss. Der Heizungsbauer baut auch nicht nur die Heizung ein, sondern schickt für jede Wartung eine Rechnung ...

>Benötigen tue ich als erstes wahrscheinlich ein Raspberry, mein Nas von Synology kann die Aufgabe nicht übernehmen?

Wenn Du keine GPIO-Ports des Raspberry Pi benötigst, würde das funktionieren, wenn auf der Synology Java 11 Läuft. Für die Heizstäbe brauchst Du aber wahrscheinlich die GPIOs und damit einen Raspberry Pi.

>Die Ersteinrichtung erfolgt wahrscheinlich direkt über das Rapberry, kann man nach der Installation dann über das Netzwerk darauf zugreifen oder muss man immer vor Ort an dem Raspberry arbeiten?

Der Raspberry Pi ist ein normaler Linux-Computer. Das Internet basiert mehr oder weniger darauf, d.h. natürlich kommst Du via Netzwerk auf den Raspberrry Pi nach der Installation.

>Als Heizstab ginge ja dann ein Model was PWM gesteuert ist und direkt an das Raspberry angeschlossen wird?

Ja, das Feature ist noch recht neu und ausser mir hat das wohl noch niemand getestet.

>Welche Wallbox wäre am geeignetsten um stufenlos zu laden?

Jede, solange sie [vom SAE unterstützt wird](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/EVCharger_DE.md).

>Wie gesagt bin ich kein PC Neuling aber habe doch Respekt vor diesem Projekt und möchte Fragen ob ich hier im Forum Unterstützung bekommen kann oder es besser lassen sollte?

Das wird Dir niemand beantworten können - diese Antwort musst Du die selbst geben.

>Ich hätte kein Problem eine Firma damit zu beauftragen aber erstens ist es sehr schwierig überhaupt jemanden zu finden, der dieses Jahr noch Zeit hat aber was viel schwieriger ist, das sich jemand mit solchen Steuerungen auskennt.

Das ist leider so: Die Wechselrichter-Hersteller verkaufen nur ihre Hardware, der Solarteur installiert das, aber um eine Integration in eine Smart-Home-Lösung kümmert sicht niemand. Das sieht auch bei jedem anders aus, d.h. wenn man das beauftragt, wird das auch schnell teuer. Wenn Du nicht so weit weg wohnen würdest, könnte ich Dir kommerzielle Hilfe anbieten.

**Comment by TimBoWi** [comment]
Hey Call777,

Bzgl. Wallbox schau dir mal die MyEnergi Zappy an. Die kannst du für PV-Überschuss nehmen. Du kannst dort mit CT-Klemmen den Überschuss ermitteln. Musst dann nur ein 3-paariges Kabel parallel zum Stromkabel einziehen oder ein Netzwerkkabel. 

Klappt ganz ohne Bastelei. 



Ansonsten die OpenWB, die ist auch super. Vielleicht wird hier perspektivisch auch ein Upgrade für Bi-Direktionales laden angeboten.

**Comment by Trunks1982** [comment]
Ich kann dir den Go E Charger empfehlen mit Phasenumschaltung.

**Comment by Call777** [comment]
Hallo guten Morgen, erstmal vielen Dank für eure zeitnahen Antworten auf meine Fragen.
Ich beschäftige mich schon seit Wochen mit diesem Projekt und habe erstmal angefangen über den Begriff Energie Management zu Googeln. 
Erstmal ging es nur um die Heizstäbe und jetzt kommt noch das E-Auto hinzu. Die Möglichkeiten am Markt sind nicht all zu gross und nach einigen Telefonaten mit Herstellerfirmen und Betrieben in meiner Nähe war ich schon fast am Kapitulieren.
Keine Zeit, noch nie was davon gehört, das Gerät ist damit nicht kompatibel und ein anderes nicht mit diesem usw. waren die meisten Aussagen.
SEMP, Modbus, Wire Card, usw. haben mich fast erschlagen weil ich das noch nie gehört habe.
Ich hatte gestern viel Zeit mich mit dem Thema SAE zu beschäftigen bzw. viel zu lesen und mansche Fragen die ich gestern noch gestellt hatte sind jetzt klarer für mich.
Auch der Zusammenhang der Steuerung und diese Software ergibt jetzt einen Sinn in meinem Kopf.
Ich habe mich dann auch gestern Nachmittag dazu entschieden das Projekt auf Basis des SAE an zu gehen.
Den Go e Charger hatte ich auch favorisiert bin dann aber auf einen negativen Testbericht gestoßen und habe mich für den Keba KeContact P30 entschieden.
Das Bidirektionale Laden wird zukünftig nur über eine DC Box laufen (Aussage von VW), das heißt ich werde dann irgendwann eh wieder eine andere Box benötigen.
Der Raspberry Pi4 Model B sowie den Sunny Home Manager 2.0 habe ich ebenfalls gekauft.
Die Heizstäbe sind aktuell nirgends verfügbar bzw. haben eine enorme Lieferzeit.
Sobald ich nun die Vorbereitungen im Keller abgeschlossen habe und ich zeitnah eine Elektrofirma finde welche alles einbaut und anschließt, werde ich mein Glück versuchen.
Raspberry aufsetzen und erstmal die Wallbox ins Sunny Portal zu bringen.
@Axel, ja wirklich schade das du anscheinend doch so weit weg bist, einer gewisse kommerzieller Hilfe wäre ich nicht abgeneigt, zumal ich dann vor Ort auch viel lernen würde.
Eine Spende geht meinerseits auf jeden Fall an dich wenn das so läuft wie geplant, das hoffe ich doch sehr und bin optimistisch.
Es wäre vorab schön wenn du dir auf er Seite der technischen Alternative die Anschlussmöglichkeiten des stufenlose Regelbaren Heizstab EHS-R anschauen könntest und mir sagst wo ich die am Raspberry anschließen muss.
Das Projekt wird sich mit Sicherheit länger hinziehen, allein schon wegen der Lieferzeit der Heizstäbe.
Ich hoffe weiterhin auf eure Unterstützung, werde anhand der Anleitungen mal schauen wie weit ich komme und melde mich bei Bedarf wieder.
Da gehe ich auf jeden Fall davon aus…..
Nochmals vielen Dank bis hierhin und ich wünsche euch ein sonniges Wochenende, Gruss Stefan.

**Reply by Trunks1982** [reply]
Hast dir schonmal mypv angeguckt dieser arbeitet direkt mit sma zusammen.😉

**Reply by silabs-RaoulvB** [reply]
> Den Go e Charger hatte ich auch favorisiert bin dann aber auf einen negativen Testbericht gestoßen und habe mich für den Keba KeContact P30 entschieden.

Hi, bin dabei eine ähnliche Entscheidung zu treffen. Die KEBAs gibt es aber in viele Varianten mit extreme Preisspanne. Darf ich fragen welche genau Du gewählt hast?

**Comment by Call777** [comment]
Ja da hast du Recht die hatte ich auch schon auf dem Schirm wegen dem SEMP Protokoll. Die kosten aber auch das Doppelte und das wäre meine Alternative zu den oben genannten.
Ich dachte mir halt wenn ich schon den Raspberry anschaffe und das PWM Signal nutzbar ist, sollte ich das auch versuchen bzw. wäre es ein Gewinn für mich.
Das würde natürlich nur mit eurer Unterstützung funktionieren, wenn Axel von vorne herein sagt das dies nicht oder nur bedingt möglich ist scheiden diese aus.

**Comment by Call777** [comment]
Java 11 läuft nun auf meinem NAS, wie kann ich den SAE darauf installieren, sehe ich eine Art Bild/Oberfläche vom SAE nach der Installation auf dem NAS? Ich würde es einfach gerne ausprobieren und mich für eine Varinte eintscheiden wollen.
Bin mir gar nicht sicher ob das so einfach möglich ist weil das NAS ja ein eigenes Betriebssystem hat und ich ja auch nur über den Browser darauf zugreife um über die Bedienoberfläche was zu ändern. Im Paketzentrum vom NAS besteht dir Möglichkeit der manuellen Installation.

**Reply by camueller** [reply]
Siehe https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/ManualInstallation_DE.md

**Comment by camueller** [comment]
> @axel, ja wirklich schade das du anscheinend doch so weit weg bist, einer gewisse kommerzieller Hilfe wäre ich nicht abgeneigt, zumal ich dann vor Ort auch viel lernen würde. Eine Spende geht meinerseits auf jeden Fall an dich wenn das so läuft wie geplant, das hoffe ich doch sehr und bin optimistisch.

Für das Lernen müsste ich nicht unbedingt vor Ort sein, das habe ich auch mit anderen schon per Screen-Sharing etc. gemacht.
Wir können Anfang kommender Woche mal einen Video-Call machen und prüfen, ob und wie ich Dir helfen kann. Schick mir mal eine Email, dann schicke ich Dir den Link zum Video-Call. Meine Kontaktdaten findest Du hier: https://github.com/camueller/cv

**Comment by Call777** [comment]
wow, das wäre schön, mach ich direkt fertig, danke.

**Comment by Call777** [comment]
Hallo, ich konnte mein Projekt nun doch erfolgreich umsetzten.
Nach einigen nervigen, zeitaufwendigen Rückschlägen, welche aber alle auf meine Unwissenheit/Unfähigkeit/ Überlesen zurückzuführen sind, gelang es mir alles nach meinen Vorstellungen zu realisieren.
Zum einen zur Heizungsunterstützung und zum anderen zum Laden meines E-Autos.
Da ich mich sehr beschränkt mit Linux usw. auskenne habe ich mich gleich an den Erfinder und Kopf des SAE gewendet und den Support in Anspruch genommen, ich hätte es sonst nicht hin bekommen.
Ich kann mich an dieser Stelle nur für den freundlichen und netten Support bedanken, DANKE A.. und jedem nahe legen diesen zu nutzen, sollte er sich auch so wenig auskennen wie ich, es lohnt sich.
Aktuell bin ich in der Feinabstimmung der einzelnen Komponenten und lerne wie sich welche Verbraucher beim automatischen  ein/ausschalten
beim SAE Verhalten.
Wie sich auch unsere Energiewende gestalten mag und wie die Köpfe der Politik auch denken mögen, ich kann heute die Energie meiner PV Anlage so steuern wie ich es mag… und das alles voll automatisiert, dank des SAE.
Die Politik spricht von der Zukunft der Energiewende ab 20… ich bin schon dabei.
Was soll ich sagen…vielen Dank für die Unterstützung, ich bin in angekommen…in der Zukunft.
